syntax = "proto3";

service AuthService {
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenResponse);
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string token = 1;
}

message ValidateTokenRequest {
    string token = 1;
}

message ValidateTokenResponse {
    bool valid = 1;
    string username = 2;
}
